evalml/tests/component_tests/test_label_encoder.py

```diff
@@ -221,3 +221,17 @@ def test_label_encoder_with_positive_label_with_custom_indices():
     y_with_custom_indices = pd.Series(["b", "a", "a"], index=[5, 6, 7])
     _, y_transformed = encoder.transform(None, y_with_custom_indices)
     assert_index_equal(y_with_custom_indices.index, y_transformed.index)
+
+
+def test_label_encoder_categorical_boolean_values():
+    X = pd.DataFrame({})
+    # binary
+    y = pd.Series(["yes", "yes", "no", "yes"])
+    y = ww.init_series(y, logical_type="Categorical")
```

Comment:
does it matter if we init this series as `Boolean` or `Categorical`?

Reply:
The behavior is the same, but if the type is passed as categorical, we would expect it to remain categorical rather than bool

Reply:
I agree that we should probably parametrize this over both `Boolean` and `Categorical.